src/main.rs

```diff
@@ -226,5 +229,5 @@ impl<W: std::io::Write> GuessingGame<W> {
 
 fn main() -> Result<(), std::io::Error> {
     // GuessingGame::new(100, rand::thread_rng(), std::io::stdout()).start()
-    RockPaperScissorsGame::new(rand::thread_rng(), std::io::stdout()).start()
```

Comment:
dovod preco to bolo ako input je, ze potom to budes vediet testovat, lebo tomu budes vediet dat nahodny generator aky ty chces (fixny). Takze toto je trocha krok zlym smerom. Treba mat ciel za tymi refaktormi. Prvotny ciel vsetkeho takehoto by malo byt, ze chces vediet napisat testy. Potom sa mozme bavit o lepsom zapise.